OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -45,96 +48,101 @@ public class AmmoPoolInfo : ITraitInfo
 		[Desc("Time to reload per ReloadCount on airfield etc.")]
 		public readonly int ReloadDelay = 50;
 
-		[Desc("Whether or not ammo is replenished on its own.")]
-		public readonly bool SelfReloads = false;
-
-		[Desc("Time to reload per ReloadCount when actor 'SelfReloads'.")]
-		public readonly int SelfReloadDelay = 50;
-
-		[Desc("Whether or not reload timer should be reset when ammo has been fired.")]
-		public readonly bool ResetOnFire = false;
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while the AmmoPool is empty.")]
+		public readonly string EmptyCondition = null;
```

Comment:
A better option would be to grant a condition for each ammo the pool has - which would resolve supporting loading stages of units akin to the RA2 Dreadnought.

Reply:
See #13057.  As much as I want to say that building an ad-hoc special case here is nasty and this should wait until someone builds the proper plumbing, it may be worth it here:
* It is indeed more logical than an `EmptyCondition`.
* `WithTurretAmmoReloadAnimation` could be removed completely from #14036.
* #13736 already sets a precedent for hacking variables from conditions.

Reply:
I can vouch for #13736 since I'm using for months now, due to it allowing one to recreate YR/Ares (Cyclic)Gattling logics - usecases of mine currently are only chargeweapons though.